lib/bankscrap/bbva/bank.rb

```diff
@@ -130,6 +132,19 @@ def login
           'eai_password'   => @password
         }
         post(BASE_ENDPOINT + LOGIN_ENDPOINT, fields: params)
+
+        # We also need to initialize a session
+        with_headers('Content-Type' => 'application/json') do
+          post(SESSIONS_ENDPOINT, fields: {
+            consumerID: CONSUMER_ID
+          }.to_json)
+        end
+
+        # We need to extract the "tsec" header from the last response.
+        # As the Bankscrap core library doesn't expose the headers of each response
+        # we have to use Mechanize's HTTP client "current_page" method.
+        tsec = @http.current_page.response['tsec']
```

Comment:
Yes, in this case I like this solution, when I asked that I was thinking about the future: we can encounter this problem in other adapters, so maybe it's worthy to return headers and body from `bankscrap` in the next major version, even if that means to update all the adapters, because in the future, with more banks, it's going to be more difficult.